nannyml/base.py

```diff
@@ -186,6 +198,31 @@ def chunk_indices(self) -> pd.Series:
     def chunk_periods(self) -> pd.Series:
         return self.data[('chunk', 'chunk', 'period')]
 
+    def _filter(
+        self,
+        period: str,
+        metrics: Optional[List[str]] = None,
+        column_names: Optional[List[str]] = None,
+        *args,
+        **kwargs
+    ) -> Result:
+        if metrics is None:
+            metrics = [metric.column_name for metric in self.metrics]
+        if column_names is None:
```

Comment:
Not entirely sure if we should have the column names here in the abstract class or not. 

We have too few examples of 2D results for now, so we might as well stick with this for the time being :+1: